sa/saro.go

```diff
@@ -99,6 +114,12 @@ func (ssa *SQLStorageAuthorityRO) GetRegistration(ctx context.Context, req *sapb
 		// replica yet. If we get a NoRows, wait a little bit and retry, once.
 		ssa.clk.Sleep(ssa.lagFactor)
 		model, err = selectRegistration(ssa.dbReadOnlyMap.WithContext(ctx), query, req.Id)
+		if err != nil {
+			if db.IsNoRows(err) {
+				ssa.lagFactorCounter.WithLabelValues("GetRegistration", "fail").Inc()
+			}
+		}
+		ssa.lagFactorCounter.WithLabelValues("GetRegistration", "pass").Inc()
```

Comment:
This is not necessarily true: the second call may have had a non-NoRows error! Does that count as "passing"?

a) If the answer is "yes", leave this as-is
b) If the answer is "no", move this into an `else` clause on the outer `if err != nil` block

Reply:
I feel like I should revert [commit](https://github.com/letsencrypt/boulder/pull/6774/commits/c2eab65977dcb89adc773f5779cf26df316b731d) to use the logic I had implemented there. Granted there's an additional bool to keep track of, but it handled this case.

Reply:
Having it in an "else" would handle this case just fine (i.e. I agreed with Samantha's comment to remove the extra var). And I truly don't know whether we want to treat a non-NoRows error as passing! I think there are good arguments both ways.